crates/napi/src/lib.rs

```diff
@@ -302,19 +312,40 @@ fn select_custom<'b>(
   builder.select(file_type)
 }
 
-fn find_files_with_lang(paths: Vec<String>, lang: &FrontEndLanguage) -> Result<WalkParallel> {
+fn find_files_with_lang(paths: Vec<String>, language_globs: Option<LanguageGlobs>, lang: &FrontEndLanguage) -> Result<WalkParallel> {
   if paths.is_empty() {
     return Err(anyhow!("paths cannot be empty.").into());
   }
 
   let mut types = TypesBuilder::new();
   let types = types.add_defaults();
+
+  let mut custom_file_type = vec![];
+  if let Some(lgs) = language_globs {
+    for (l, globs) in lgs {
+      let result = SupportLang::from_str(&l);
+      if result.is_err(){
+        return Err(anyhow!("unrecognized language in language globs").into());
+      }
+      let result = FrontEndLanguage::try_from(result.unwrap());
+      match result {
+        Ok(value) => {
+          if value == *lang{
+            custom_file_type.extend(globs);
+          }
+        },
+        Err(e) => return Err(anyhow!(format!("{} is not supported in napi", e)).into()),
+      }
+    } 
+  }
+
+  let custom_file_type: Vec<&str> = custom_file_type.iter().map(|s| s.as_str()).collect();
   let types = match lang {
-    FrontEndLanguage::TypeScript => select_custom(types, "myts", &["*.ts", "*.mts", "*.cts"]),
-    FrontEndLanguage::Tsx => select_custom(types, "mytsx", &["*.tsx", "*.mtsx", "*.ctsx"]),
-    FrontEndLanguage::Css => types.select("css"),
-    FrontEndLanguage::Html => types.select("html"),
-    FrontEndLanguage::JavaScript => types.select("js"),
+    FrontEndLanguage::TypeScript => select_custom(types, "myts", vec!["*.ts", "*.mts", "*.cts"], custom_file_type),
+    FrontEndLanguage::Tsx => select_custom(types, "mytsx", vec!["*.tsx", "*.mtsx", "*.ctsx"], custom_file_type),
+    FrontEndLanguage::Css => select_custom(types, "css", vec!["css"], custom_file_type),
+    FrontEndLanguage::Html => select_custom(types, "html",  vec!["html"], custom_file_type),
```

Comment:
Can we use the same configuration here?

https://github.com/ast-grep/ast-grep/blob/main/crates/language/src/lib.rs#L288-L313